### PR TITLE
cmd/search/main: Handle pre-existing build-log.txt

### PR DESCRIPTION
**Pull request by wking**
With master, we're getting errors like:

```
W0426 17:39:57.457197       1 main.go:177] Job index failed: unable to index prow jobs from Deck, could not create log file: open /var/lib/ci-search/origin-ci-test/logs/periodic-ci-azure-vmimage/3/build-log.txt: file exists
```

With this commit, I open `build-log.txt` before making the expensive HTTP call, to use creation as a lock.  If we fail for most reasons, I make a best-effort attempt to remove the file so the next attempt will try again.  For 404s and other permanently-fatal errors, I leave the empty file on disk so future workers don't repeat the known-to-not-work HTTP request.

Workers finding an existing file on disk return nil now ("it's there"), making `fetchJob` idempotent.

There are a few possible races here (e.g. the search process SIGKILLed before completely filling in the local file) that could lead to a busted on-disk file.  But these only live 24 hours anyway, so having the occasional busted `build-log.txt` on the local disk in the short term is acceptable.

### CODE REVIEW

File: cmd/search/main.go [hunk 1]
```diff
@@ -284,12 +284,27 @@ func fetchJob(client *http.Client, job *ProwJob, indexedPaths *pathIndex, toDir
 		return nil
 	}
 
+	pathOnDisk := filepath.Join(append([]string{toDir}, strings.Split(logPath, "/")...)...)
```
Comment (smarterclayton) [inline cmd/search/main.go]: This is a bug in something with indexedPaths.  IndexedPaths.MetadataFor should return true if the file exists, so if we're getting here that means there's a mismatch between the path we write and the path we index.  The path we index should be `/origin-ci-test/logs/....`

Comment (wking) [reply cmd/search/main.go]: > ...there's a mismatch between the path we write and the path we index...

Maybe #17 fixes this, then?

Comment (wking) [reply cmd/search/main.go]: /hold

while we sort out wether #17 is good enough.

**Comment by wking** [comment]
Obsoleted by #22.

**Pull request closed without merging**